src/context_free.ml

```diff
@@ -461,20 +461,22 @@ class map_top_down ?(expect_mismatch_handler=Expect_mismatch_handler.nop)
        - func.pexp_desc = Pexp_ident _
     *)
     method private pexp_apply_without_traversing_function base_ctxt e func args =
-      let { pexp_desc = _; pexp_loc; pexp_attributes } = e in
+      let { pexp_desc = _; pexp_loc; pexp_attributes; pexp_loc_stack = _ } = e in
       let func =
-        let { pexp_desc; pexp_loc; pexp_attributes } = func in
+        let { pexp_desc; pexp_loc; pexp_attributes; pexp_loc_stack } = func in
         let pexp_attributes = self#attributes base_ctxt pexp_attributes in
         { pexp_desc
         ; pexp_loc (* location doesn't need to be traversed *)
         ; pexp_attributes
+        ; pexp_loc_stack
         }
       in
       let args = List.map args ~f:(fun (lab, exp) -> (lab, self#expression base_ctxt exp)) in
       let pexp_attributes = self#attributes base_ctxt pexp_attributes in
       { pexp_loc
       ; pexp_attributes
       ; pexp_desc = Pexp_apply (func, args)
+      ; pexp_loc_stack = []
```

Comment:
Why are you dropping the loc stack here?

Reply:
Fixed.